### PR TITLE
Added 'setFont' and 'setAnchorPoint' to CCEditBox

### PR DESCRIPTION
**Pull request by neokim**
Now CCEditBox has an ability to change its font and size, and also it is working fine now with 'setAnchorPoint' on iOS which wasn't working properly.
The changes are applied to the bundle test project and tested on iOS and Android.
ps) Because there was a linkage error on Android build with the test project with its default setting, I needed to add 'APP_ABI = armeabi-v7a' to Application.mk. I'm using ndk-r7b.

### CODE REVIEW

File: samples/Cpp/TestCpp/Classes/ExtensionsTest/EditBoxTest/EditBoxTest.cpp [hunk 1]
```diff
@@ -38,8 +38,14 @@
     // top
     m_pEditName = CCEditBox::create(editBoxSize, CCScale9Sprite::create("extensions/green_edit.png"));
     m_pEditName->setPosition(ccp(visibleOrigin.x+visibleSize.width/2, visibleOrigin.y+visibleSize.height*3/4));
+#ifdef CC_TARGET_OS_IPHONE
```
Comment (dumganhar) [inline samples/Cpp/TestCpp/Classes/ExtensionsTest/EditBoxTest/EditBoxTest.cpp]: I think using `#if (CC_TARGET_PLATFORM == CC_PLATFORM_IOS)` is better.

File: scripting/javascript/bindings/generated [hunk 2]
```diff
@@ -1 +1 @@
-Subproject commit 35c9124a808c8e67b5a9bef68d4ff4433dad8bd4
+Subproject commit 4731abe94375ad4b9bde20cfd8bd48d0e1538fb1
```
Comment (dumganhar) [inline scripting/javascript/bindings/generated:1]: Why did you change the submodule reference?

File: tools/cxx-generator [hunk 3]
```diff
@@ -1 +1 @@
-Subproject commit b92f99a3e53c5d4c19eb166e649664bd7fd80bab
+Subproject commit 39286b435de49cfb745297bc514ed9f7d6d1f7f9
```
Comment (dumganhar) [inline tools/cxx-generator:1]: Why did you change the submodule reference?

File: samples/Cpp/TestCpp/proj.android/jni/Application.mk [hunk 4]
```diff
@@ -1,2 +1,3 @@
 APP_STL := gnustl_static
 APP_CPPFLAGS := -frtti -DCC_ENABLE_CHIPMUNK_INTEGRATION=1 -DCOCOS2D_DEBUG=1
+APP_ABI = armeabi-v7a
```
Comment (dumganhar) [inline samples/Cpp/TestCpp/proj.android/jni/Application.mk:3]: Which host os you are using? MAC or Windows? I have tested on MAC, and it was ok.

File: samples/Cpp/TestCpp/Classes/ExtensionsTest/EditBoxTest/EditBoxTest.cpp (hunk 1 above)
Comment (neokim) [inline samples/Cpp/TestCpp/Classes/ExtensionsTest/EditBoxTest/EditBoxTest.cpp]: No problem.

File: scripting/javascript/bindings/generated (hunk 2 above)
Comment (neokim) [inline scripting/javascript/bindings/generated:1]: I was new to git in the morning and still I am. I don't know what they meant, just thought that it's kind of repository metadata. You can just rollback them if they are not necessary.

File: samples/Cpp/TestCpp/proj.android/jni/Application.mk (hunk 4 above)
Comment (neokim) [inline samples/Cpp/TestCpp/proj.android/jni/Application.mk:3]: Mac. It might be caused by ndk version. I'm using ndk-r7b. What about you?

Comment (dumganhar) [inline samples/Cpp/TestCpp/proj.android/jni/Application.mk:3]: Yes, please use ndk-r8d and try again.

File: extensions/GUI/CCEditBox/CCEditBoxImplAndroid.cpp [hunk 5]
```diff
@@ -88,6 +90,19 @@ bool CCEditBoxImplAndroid::initWithSize(const CCSize& size)
...
+	if(m_pLabel != NULL) {
+		m_pLabel->setFontName(pFontName);
+		m_pLabel->setFontSize(fontSize);
+	}
+	
+	if(m_pLabelPlaceHolder != NULL) {
+		m_pLabelPlaceHolder->setFontName(pFontName);
+		m_pLabelPlaceHolder->setFontSize(fontSize);
+	}
+}
```
Comment (dumganhar) [inline extensions/GUI/CCEditBox/CCEditBoxImplAndroid.cpp:104]: How about separating `setFont` to methods as follows?
- setFontName
- setPlaceholderFontName
- setFoneSize
- setPlaceholderFontSize

Comment (neokim) [inline extensions/GUI/CCEditBox/CCEditBoxImplAndroid.cpp:104]: That was my first bet. But then, on iOS `[UIFont fontWithName:fntName size:fontSize]` needs to be called twice to change both font and size.

Comment (dumganhar) [inline extensions/GUI/CCEditBox/CCEditBoxImplAndroid.cpp:104]: OK, how about
- setFont
- setPlaceholderFont
  ?

Comment (neokim) [inline extensions/GUI/CCEditBox/CCEditBoxImplAndroid.cpp:104]: I've added `setPlaceholderFont()` API to CCEditBox. It is not implemented on iOS because `UITextField` of iOS doesn't provide any way to customize placeholder. To do that, `UITextField` needs to be subclassed and overridden.

File: samples/Cpp/TestCpp/Classes/ExtensionsTest/EditBoxTest/EditBoxTest.cpp (hunk 1 above)
Comment (dumganhar) [inline samples/Cpp/TestCpp/Classes/ExtensionsTest/EditBoxTest/EditBoxTest.cpp]: Could you also update this?

File: extensions/GUI/CCEditBox/CCEditBoxImplAndroid.cpp (hunk 5 above)
Comment (dumganhar) [inline extensions/GUI/CCEditBox/CCEditBoxImplAndroid.cpp:104]: > To do that, UITextField needs to be subclassed and overridden.

OK, thank you. So do you have a plan to make it supported on iOS?

Comment (neokim) [inline extensions/GUI/CCEditBox/CCEditBoxImplAndroid.cpp:104]: I have customized the iOS implementation of CCEditBox in my project to solve the issue #1971 and some other problems. It already supports placeholder customization partially.
The customization I made is that UITextField is visible only when the CCEditBox is in edit mode. And in non-edit mode, UITextField is hidden and CCLabelTTF is shown instead with all same attributes at the same place. So the placeholder in non-edit mode can be customized.
I think that this customization is just a work-around for some problems and definitely not a perfect solution.
But it is still useful, so I'm going to issue another pull request for this after current pull request.

Comment (dumganhar) [inline extensions/GUI/CCEditBox/CCEditBoxImplAndroid.cpp:104]: OK, I will merge this Pull Request today. Thanks @neokim .

**Comment by dumganhar** [comment]
Seems win32 and mac platforms also need to be updated.:)

**Comment by dumganhar** [comment]
Merged at https://github.com/cocos2d/cocos2d-x/pull/2044
Thanks.
